### PR TITLE
Exclude mimetypelist.js from signing/integrity check

### LINKED ISSUE #29277: Integrity check fail 10.0.4RC1 on mimetype file
Install 10.0.4beta1 and check the integrity warnings.

The file core/js/mimetypelist.js must not be checked at all, but somehow a signature got generated for it.

It seems only half of the code was adjusted for exclusion. Also need to adjust the code that generates the signature.

This regression introduced through https://github.com/owncloud/core/pull/28973

### PR DESCRIPTION
**Pull request by VicDeo**
## Description
Adds regex exclusions by `path/filename` into ExcludeFileByNameFilterIterator

## Related Issue
Fixes https://github.com/owncloud/core/issues/29277

## Motivation and Context
Signing `core/js/mimetypelist.js`

## How Has This Been Tested?
Unit-tested

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: lib/private/IntegrityCheck/Checker.php [hunk 1]
```diff
@@ -179,22 +179,6 @@ private function generateHashes(\RecursiveIteratorIterator $iterator,
 				continue;
 			}
 
-			$exclusionList = [
-				'/core/js/mimetypelist.js'
-			];
-
-			$found = false;
-			foreach ($exclusionList as $entry) {
```
Comment (PVince81) [inline lib/private/IntegrityCheck/Checker.php]: you likely still need to keep something here to prevent generating the hashes in the first place ?

Comment (VicDeo) [reply lib/private/IntegrityCheck/Checker.php]: @PVince81 `generateHashes` takes a folder iterator. I excluded this path from the items returned by this iterator.

Comment (VicDeo) [reply lib/private/IntegrityCheck/Checker.php]: It's reused everywhere, so I basically made `core/js/mimetypelist.js` completely invisible for signing code. See 
https://github.com/owncloud/core/blob/8c8c3542ecbdf0420b71cc2458d3c0bd8d70456b/lib/private/IntegrityCheck/Checker.php#L157
https://github.com/owncloud/core/blob/8c8c3542ecbdf0420b71cc2458d3c0bd8d70456b/lib/private/IntegrityCheck/Checker.php#L386

**Review by PVince81 — APPROVED** [review]
:+1: thanks for clarifying

**Comment by PVince81** [comment]
@VicDeo please backport to stable10

**Comment by VicDeo** [comment]
Stable10: https://github.com/owncloud/core/pull/29316

**Pull request merged**